SDWebImage/Private/SDAsyncBlockOperation.m

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+#import "SDAsyncBlockOperation.h"
+
+@interface SDAsyncBlockOperation ()
+
+@property (nonatomic, assign) BOOL isExecuting;
+@property (nonatomic, assign) BOOL isFinished;
+@property (nonatomic, copy, nonnull) SDAsyncBlock executionBlock;
+
+@end
+
+@implementation SDAsyncBlockOperation
```

Comment:
@dreampiggy Any reason to create this new class?

Reply:
This class is used, to simply create a `NSOperation`, which can make a async completion callback into the `NSOperation` and get benefit from the built-in `NSOperationQueue`. If you want to use the some power feature (like the `maxConcurrentOperationCount`, the dependency manage) from the high level API instead of using the GCD.

The `NSBlockOperation` can only been used when your code in the block, is sync. The `NSBlockOperation` will be marked as finished after your last line of code is returned. So it's not help for async method call (always return immediatelly).

This `SDAsyncBlockOperation` will be marked as finished after you call `complete` inside the async block, So it's just a easy to use our manager's API `loadImageWithURL:completed:`

Reply:
The IGListKit private headers: https://github.com/Instagram/IGListKit/tree/master/Source/Internal. We follow the same structure as they are.

IGListKit is used widely in the World (even widely than SDWebImage itself) and I believe facebook team have a strong knowledge about the Project management, code structure and the build tools (CocoaPods). This is the reason.

Reply:
@zhongwuzw You can see issue [#2630](https://github.com/SDWebImage/SDWebImage/issues/2630).

Reply:
@dreampiggy @kinarob This feature has severe bugs, would leads operation queue hung up. I'll fix them.